app/controllers/user_controller.rb

```diff
@@ -0,0 +1,47 @@
+class UserController < ApplicationController
+  get '/' do
+    erb :'users/index'
+  end
+
+  get '/signup' do
+    redirect '/tweets' if session.key?(:user_id)
+    erb :'users/new'
+  end
+
+  post '/signup' do
+    redirect '/signup' if params[:username].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.